### DISCUSSION TITLE
Custom icons from svg files?

### DISCUSSION THREAD

**Discussion by t1m4lc** [ng-icons/ng-icons#119] [Q&A] [answered]
It is possible to add custom icons ? If yes how to do it ?

**Comment by ashley-hunter** [comment, marked as answer]
Hi! Absolutely! The icons we provide are just SVG strings, so you can add your own very easily:

```ts
const file = `<svg ...></svg>`;

...

@Component({
  viewProviders: [
      provideIcons({ file })
  ]
})
```

now you can use this in your template:

```html
<ng-icon name="file" />
```

That should allow you to use your custom icons. By default our icons inherit the color of the text, so if you wish to preserve this functionality, you should update any uses of fill or stroke-color to have a value of currentColor.

I hope that helps!
